draft-irtf-cfrg-rsa-blind-signatures.md

```diff
@@ -614,42 +520,28 @@ fails to verify. However, if an attacker can coerce the client to use these inva
+Based on this fact, using the core protocol functions in {{core-protocol}} is possibly unsafe,
 unless one of the following conditions are met:
 
 1. The client has proof that the signer's public key is honestly generated. {{GRSB19}} presents
   some (non-interactive) honest-verifier zero-knoweldge proofs of various statements about the
   public key.
 2. The client input message has high entropy.
 
-The interface in {{salted-interface}} is designed to explicitly inject fresh entropy alongside
-each message to satisfy condition (2). As such, this interface is safe for all application use
-cases.
+The named variants that use message randomization -- RSABSSA-SHA384-PSS-Randomized and
+RSABSSA-SHA384-PSSZERO-Randomized -- explicitly inject fresh entropy alongside each message
+to satisfy condition (2). As such, these variants are safe for all application use cases.
 
-Note that this interface effectively means that the resulting signature is always randomized.
+Note that these variants effectively mean that the resulting signature is always randomized.
 As such, this interface is not suitable for applications that require deterministic signatures.
-See {{det-sigs}} for more details.
 
-## Randomized and Deterministic Signatures {#det-sigs}
+## PSS Salt Choice
 
-When sLenInBytes > 0, the PSS salt is a randomly generated string chosen when a message is encoded.
-This means the resulting signature is non-deterministic. As a result, two signatures over
-the same message will be different. If the salt is not generated randomly, or is otherwise
-constructed maliciously, it might be possible for the salt to encode information that is
-not present in the signed message. For example, the salt might be maliciously constructed
-to encode the local IP address of the client. As a result, APIs SHOULD NOT allow clients
-to provide the salt directly; see {{apis}} for API considerations.
-
-When sLenInBytes = 0, the PSS salt is empty and the resulting signature is deterministic. Such
-signatures may be useful for applications wherein the only desired source of entropy is
-the input message. Note, however, that this can be unsafe if the input message does not have
-sufficient entropy; see {{message-entropy}} for more details.
-
-Applications that use deterministic signatures SHOULD carefully analyze the security
-implications, taking into account the possibility of adversarially generated signer
-keys as described in {{message-entropy}}. When it is not clear whether an application
-requires deterministic or randomized signatures, applications SHOULD use randomized
-signatures, and the salted interface described in {{salted-interface}} SHOULD be used.
+For variants that have a non-zero PSS salt, it is important that this salt is generated randomly.
+If the PSS salt is not generated randomly, or is otherwise constructed maliciously, it might be
+possible for the salt to encode information that is not present in the signed message. For example,
```

Comment:
I found the example not really useful. Better just mention that the salt must be generated using a cryptographic PRNG.
Also, it isn't the case that the salt is derived from the message?, if so, this paragraph can be shortened.

Reply:
The salt isn't derived from the message. This example demonstrates why it's important to generate the salt randomly, so I think we'll keep it.